components/rmrk/Gallery/History.vue

```diff
@@ -108,287 +102,242 @@ import {
-          event['From'] = newEvent['caller']
-          event['To'] = ''
-          event['Percentage'] = parseInt(newEvent['meta'])
-          break
-        case InteractionBsxOnly.PAY_ROYALTY:
-          event['From'] = newEvent['caller']
-          event['To'] = ''
-          event['Amount'] = newEvent['meta']
-          break
-        default:
-          // unsupported event
-          continue
-      }
+    // Date
+    const date = new Date(newEvent['timestamp'])
+    event['Date'] = parseDate(date)
 
-      event['Type'] = event['Type'] ?? newEvent['interaction']
+    // Time
+    event['Time'] = formatDistanceToNow(date, { addSuffix: true })
 
-      // Item
-      if (this.displayItem) {
-        event['Item'] = newEvent['nft']
-      }
+    event['Block'] = String(newEvent['blockNumber'])
 
-      // Amount
-      event['Amount'] = event['Amount'] ?? '-'
+    // ID for table: Use a unique key of your data Object for each row.
+    event['ID'] = newEvent['timestamp'] + newEvent['id']
 
-      // Date
-      const date = new Date(newEvent['timestamp'])
-      event['Date'] = parseDate(date)
-
-      // Time
-      event['Time'] = formatDistanceToNow(date, { addSuffix: true })
-
-      event['Block'] = String(newEvent['blockNumber'])
-
-      // ID for table: Use a unique key of your data Object for each row.
-      event['ID'] = newEvent['timestamp'] + newEvent['id']
-
-      // Push to chart data
-      if (newEvent['interaction'] === Interaction.LIST) {
-        this.pushChartData(chartData.list, date, event['Amount'])
-      } else if (newEvent['interaction'] === Interaction.BUY) {
-        this.pushChartData(chartData.buy, date, event['Amount'])
-      }
-
-      this.copyTableData.push(event)
-    }
-    this.copyTableData = this.copyTableData.reverse()
-    this.updateDataByEvent()
-
-    if (!this.data.length) {
-      this.event = HistoryEventType.ALL
+    // Push to chart data
+    if (newEvent['interaction'] === Interaction.LIST) {
+      pushChartData(chartData.list, date, event['Amount'])
+    } else if (newEvent['interaction'] === Interaction.BUY) {
+      pushChartData(chartData.buy, date, event['Amount'])
     }
 
-    this.$emit('setPriceChartData', [chartData.buy, chartData.list])
+    copyTableData.value.push(event)
   }
 
-  private pushChartData(array, date, amount) {
-    array.push([date, parseChartAmount(amount, this.decimals)])
-  }
+  copyTableData.value = copyTableData.value.reverse()
+  updateDataByEvent()
 
-  @Watch('events', { immediate: true })
-  public watchEvent(): void {
-    if (this.events) {
-      this.createTable()
-    }
+  if (!data.value.length) {
+    event.value = HistoryEventType.ALL
   }
 
-  @Watch('event', { immediate: true })
-  public watchInteractionEvent(): void {
-    if (this.event) {
-      this.updateDataByEvent()
-    }
-  }
+  emit('setPriceChartData', [chartData.buy, chartData.list])
 }
+
+watch(
+  () => prop.events,
+  () => {
+    createTable()
+  }
```

Comment:
```suggestion
createTable
```


not important, just style preference, feel free to resolve and ignore if you disagree